src/core/prompts/system.ts

```diff
@@ -98,6 +98,7 @@ ${getToolDescriptionsForMode(
 	experiments,
 	partialReadsEnabled,
 	settings,
+	settings?.disabledTools,
```

Comment:
use the existing settings object, getToolDescriptionsForMode most access settings.disabledTools